src/databricks/labs/ucx/source_code/linters/python_ast.py

```diff
@@ -0,0 +1,324 @@
+import abc
+import logging
+from collections.abc import Iterable
+from typing import TypeVar, Generic, cast
+
+from astroid import (  # type: ignore
+    parse,
+    Attribute,
+    Call,
+    Const,
+    Import,
+    ImportFrom,
+    Module,
+    Name,
+    NodeNG,
+)
+
+logger = logging.getLogger(__file__)
+
+missing_handlers: set[str] = set()
+
+T = TypeVar("T", bound=NodeNG)
+
+
+# disclaimer this class is NOT thread-safe
+class Tree(Generic[T]):
```

Comment:
```suggestion
class Tree:
```
It has no generic members. Only arguments for some static methods are generic